src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/AuthorisedCreateEventOperation.java

```diff
@@ -80,6 +86,14 @@ public CaseDetails createCaseEvent(String caseReference,
         return verifyReadAccess(caseTypeDefinition, accessProfiles, caseDetails);
     }
 
+    private void verifyCaseAccessCategories(Set<AccessProfile> accessProfiles, CaseDetails existingCaseDetails) {
+        Optional<CaseDetails> filteredCaseDetails = Optional.of(existingCaseDetails)
+            .filter(caseAccessCategoriesService.caseHasMatchingCaseAccessCategories(accessProfiles, false));
+        if (filteredCaseDetails.isEmpty()) {
+            throw new ResourceNotFoundException(NO_CASE_TYPE_FOUND);
```

Comment:
Maybe for this scenario we should throw something else as it's more that the user doesn't have access to the case - may make more sense to stay consistent with elsewhere e.g. [here](https://github.com/hmcts/ccd-data-store-api/blob/63b846d297153af1a9fa22b3251b15c9654d52fc/src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseController.java#L115)
```suggestion
            throw new CaseNotFoundException(caseDetails.getReferenceAsString());
```